code/modules/mob/observer/eye/blueprints_eye.dm

```diff
@@ -0,0 +1,275 @@
+#define MAX_AREA_SIZE 300
+
+/mob/observer/eye/blueprints
+	
+	var/list/selected_turfs = list() // Associative list of turfs -> boolean validity that the player has selected for new area creation.
+	var/list/selection_images = list()
+	var/turf/last_selected_turf
+	var/image/last_selected_image
+	
+	// On what Z-levels this can be used to modify or create areas.
+	var/list/valid_z_levels = list()
+
+	// Displayed to the user to allow them to see what area they're hovering over.
+	var/obj/effect/overlay/area_name_effect
+	var/area_prefix
+	
+	// Displayed to the user on failed area creation.
+	var/list/errors = list()
+
+/mob/observer/eye/blueprints/Initialize(var/mapload, var/list/valid_zls, var/area_p)
+	. = ..(mapload)
+
+	valid_z_levels = valid_zls
+	area_prefix = area_p
+
+	area_name_effect = new()
+
+	area_name_effect.maptext_height = 64
+	area_name_effect.maptext_width = 128
+	area_name_effect.layer = FLOAT_LAYER
+	area_name_effect.plane = HUD_PLANE
+	area_name_effect.appearance_flags = APPEARANCE_UI_IGNORE_ALPHA
+	area_name_effect.screen_loc = "LEFT+1,BOTTOM+2"
+
+	last_selected_image = image('icons/effects/blueprints.dmi', "selected")
+	last_selected_image.plane = OBSERVER_PLANE
+	last_selected_image.appearance_flags = NO_CLIENT_COLOR
+
+/mob/observer/eye/blueprints/Destroy()
+	. = ..()
+	QDEL_NULL(area_name_effect)
+	errors = null
+	selected_turfs = null
+	valid_z_levels = null
+	last_selected_turf = null
+
+/mob/observer/eye/blueprints/release(var/mob/user)
+	if(owner && owner.client && user == owner)
+		owner.client.images.Cut()
+	. = ..()
+	
+/mob/observer/eye/blueprints/proc/create_area()
+	var/area_name = sanitizeSafe(input("New area name:","Blueprint Editing", ""), MAX_NAME_LEN)
+	if(!area_name || !length(area_name))
+		return
+	if(length(area_name) > 50)
+		to_chat(owner, SPAN_WARNING("That name is too long!"))
+		return
+
+	if(!check_selection_validity())
+		to_chat(owner, SPAN_WARNING("Could not mark area: [english_list(errors)]!"))
+		return
+
+	var/area/A = new
+	A.SetName(area_name)
+	A.power_equip = 0
+	A.power_light = 0
+	A.power_environ = 0
+	A.always_unpowered = 0
+	for(var/turf/T in selected_turfs)
+		ChangeArea(T, A)
+	remove_selection() // Reset the selection for clarity.
+
+/mob/observer/eye/blueprints/proc/remove_area()
+	var/area/A = get_area(src)
+	if(!check_modification_validity())
+		return
+	if(A.apc)
+		to_chat(owner, SPAN_WARNING("You must remove the APC from this area before you can remove it from the blueprints!"))
+		return
+	to_chat(owner, SPAN_NOTICE("You scrub [A.name] off the blueprints."))
+	log_and_message_admins("deleted area [A.name] via station blueprints.")
+	qdel(A)
+
+/mob/observer/eye/blueprints/proc/edit_area()
+	var/area/A = get_area(src)
+	if(!check_modification_validity())
+		return
+	var/prevname = A.name
+	var/new_area_name = sanitizeSafe(input("New area name:","Blueprint Editing", prevname), MAX_NAME_LEN)
```

Comment:
Same as before, instead of "Blueprint Editing"  "Area Editing" would be more contextually appropriate.